src/main/java/peoplesoft/logic/commands/CommandHelpMessage.java

```diff
@@ -0,0 +1,103 @@
+package peoplesoft.logic.commands;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javafx.beans.property.SimpleStringProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+/**
+ * Handle help instructions for commands.
+ */
+public class CommandHelpMessage {
+    public static final String PROPERTY_DELIMITER = "----";
+    public static final String MESSAGE_DELIMITER = "--------";
+
+    private final SimpleStringProperty command;
+    private final SimpleStringProperty format;
+    private final SimpleStringProperty examples;
+
+
+    /**
+     * Constructor for CommandHelpMessage.
+     *
+     * @param command command word
+     * @param format format descriptor of command
+     * @param examples example usage of command
+     */
+    public CommandHelpMessage(String command, String format, String examples) {
+        this.command = new SimpleStringProperty(command);
+        this.format = new SimpleStringProperty(format);
+        this.examples = new SimpleStringProperty(examples);
+    }
+
+    public String getCommand() {
+        return command.get();
+    }
+
+    public String getFormat() {
+        return format.get();
+    }
+
+    public String getExamples() {
+        return examples.get();
+    }
+
+    /**
+     * Converts a command's help message into string delimited by the property delimiter.
+     *
+     * @return formatted help message.
+     */
+    public String toString() {
+        return getCommand() + PROPERTY_DELIMITER
+                + getFormat() + PROPERTY_DELIMITER
+                + getExamples();
+    }
+
+    /**
+     * Convert list of commands to string.
+     *
+     * @param commands list of commands
+     * @return string delimited by -------- and ----
+     */
+    public static String asString(ObservableList<CommandHelpMessage> commands) {
+        StringBuilder parsedCommandMessages = new StringBuilder();
+
+        for (CommandHelpMessage msg: commands) {
+            parsedCommandMessages.append(msg);
+            parsedCommandMessages.append(MESSAGE_DELIMITER);
+        }
+
+        return parsedCommandMessages.toString();
+    }
+
+    /**
+     * Parse a string into an ObservableList of CommandHelpMessages.
+     *
+     * @param str string to be parsed
+     * @return ObservableList of help messages for commands in string.
+     */
+    public static ObservableList<CommandHelpMessage> asObservableList(String str) {
```

Comment:
what's this for?

Reply:
I had something like [this](https://docs.oracle.com/javafx/2/ui_controls/table-view.htm) in mind for the implementation of TableView since the current feedback to the user is a concatenated string of all command messages. From what I understand of TableView, it seems useful to have a method to parse the string back into the original`<CommandHelpMessage>` list  so that elements of the table can be set with `table.setItems(list)`

If you have a different structure/implementation in mind, do feel free to remove this

Reply:
i think it would be better to just show the user the help window when we get a `help` command, since that already has a table with which we can show all the commands/formats/examples?

i'm not quite sure i like the idea of turning some text into a human-readable pretty-printed format just to have to parse it back again, because it feels like it'll be rather prone to breakage